### PR TITLE
Allow a direct connection to postgres from webserver for LISTEN/NOTIFY channel

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

pgBouncer works best in Transaction pooling mode with our aiopg backend. BUT, this mode does not allow for listening to postgres notifications see [here](https://www.pgbouncer.org/features.html)

therefore, a direct connection to postgres is allowed for the webserver service ONLY to this end. everything else seems to work as expected until now.

- changed the docker container used for pgbouncer as it was deprecated... now using edoburu/pgbouncer:1.15.0@sha256:2f47bf272fa9fdf25c100d11f1972b23af61a351a136d3721bfa6bdb52630426
- increase max_client_conn to 10000 on pgbouncer to allow for err... 10000 clients... not the default 100...
- POSTGRES_LONG_RUNNING_SESSION_HOST env variable added to .env-devel for direct connections with postgres DB
-   (⚠️ devops) due to change of docker image container: please change ENVs to DB_HOST

NOTE:
To access pgbouncer statistics:
```console
docker exec -it PGBOUNCER_ID sh
psql postgres://127.0.0.1/pgbouncer -U scu
# enter postgres password
show help;
```

### CODE REVIEW

File: services/web/server/src/simcore_service_webserver/computation_comp_tasks_listening_task.py [hunk 1]
```diff
@@ -165,12 +173,39 @@ async def listen(app: web.Application):
...
+async def listening_channel_pg_engine(app: web.Application) -> Engine:
+    cfg = app[APP_CONFIG_KEY][CONFIG_SECTION_NAME]
+    pg_cfg = cfg["postgres"]
+
+    dsn = DataSourceName(
+        application_name=f"{__name__}_{id(app)}",
+        database=pg_cfg["database"],
+        user=pg_cfg["user"],
+        password=pg_cfg["password"],
+        host=pg_cfg["long_running_session_host"],
```
Comment (GitHK) [inline services/web/server/src/simcore_service_webserver/computation_comp_tasks_listening_task.py:185]: so here only 1 connection will be acquired form the pool, correct?

**Review by GitHK — APPROVED** [review]
Very nice, thanks for sharing the info on the issue.

File: services/web/server/src/simcore_service_webserver/computation_comp_tasks_listening_task.py (hunk 1 above)
Comment (sanderegg) [reply services/web/server/src/simcore_service_webserver/computation_comp_tasks_listening_task.py:185]: yes, it's a pool with only one connection anyway. but it has to stay open the whole time.

File: tests/e2e/utils/wait_for_services.py [hunk 2]
```diff
@@ -24,7 +24,7 @@
 current_dir = Path(sys.argv[0] if __name__ == "__main__" else __file__).resolve().parent
 
 WAIT_BEFORE_RETRY = 10
-MAX_RETRY_COUNT = 10
+MAX_RETRY_COUNT = 20
```
Comment (pcrespov) [inline tests/e2e/utils/wait_for_services.py:27]: what did you observe to double the retries?

File: services/web/server/src/simcore_service_webserver/computation_comp_tasks_listening_task.py [hunk 3]
```diff
@@ -165,12 +173,40 @@ async def listen(app: web.Application):
...
+        port=pg_cfg["port"],
+    )
+    MIN_SIZE = 1
```
Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/computation_comp_tasks_listening_task.py:188]: MINOR: why not directly in ``minsize=1, maxsize=1``?

File: services/web/server/src/simcore_service_webserver/computation_comp_tasks_listening_task.py [hunk 4]
```diff
@@ -165,12 +173,40 @@ async def listen(app: web.Application):
...
+    MAX_SIZE = 1
+    log.info("Creating listening pg engine for %s", dsn)
+    for attempt in Retrying(**PostgresRetryPolicyUponInitialization(log).kwargs):
+        with attempt:
+            engine = await create_pg_engine(dsn, minsize=MIN_SIZE, maxsize=MAX_SIZE)
```
Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/computation_comp_tasks_listening_task.py:193]: shouldn't we make sure there is a SINGLE engine per application? 
- the db app submodule could create them upon startup and is responsible of closing them as well ( e.g. appending to ``app.cleanup_ctx`` )
- See ``setup`` in ``services/web/server/src/simcore_service_webserver/db.py``
- This way, we can monitor the status of all connections to the db, either direct or via the pgbouncer

**Review by pcrespov — CHANGES_REQUESTED** [review]

File: tests/e2e/utils/wait_for_services.py (hunk 2 above)
Comment (sanderegg) [reply tests/e2e/utils/wait_for_services.py:27]: e2e CI failing 1/4 of the times because 1 sidecar was not yet started

**Review by pcrespov — APPROVED** [review]

File: services/web/server/src/simcore_service_webserver/computation_comp_tasks_listening_task.py (hunk 4 above)
Comment (sanderegg) [reply services/web/server/src/simcore_service_webserver/computation_comp_tasks_listening_task.py]: done

**Pull request merged**